rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -328,7 +328,7 @@ public int leave(URI lraId, String compensatorUrl) {
         }
     }
 
-    public synchronized int joinLRA(StringBuilder recoveryUrl, URI lra, long timeLimit,
+    public int joinLRA(StringBuilder recoveryUrl, URI lra, long timeLimit,
```

Comment:
As we discussed offline, I'm not sure what all the possible implications of removing synchronized from this method could be. I would like to hear from @mmusgrov why it was synchronized in the first place.

Reply:
Good idea, I put the on hold label.

Reply:
@marcosgopen I added a comment to the JBTM indicating that the tests are good but I'm not convinced that removing the monitor is the correct approach. Personally I would keep the tests but withdraw the fix and work on a different fix for the lock contention, do you mind if I look into the alternate approach?